### PR TITLE
[DUOS-918][risk=low] datasetapi to service

### PR DESCRIPTION
**Pull request by rfricke-asymmetrik**
**Addresses**
https://broadworkbench.atlassian.net/browse/DUOS-918

**Changes**

- Removed the DatasetAPI Singleton classes
- Moved DatasetAPI methods to DatasetService
- Updated references to DatasetAPI to DatasetService

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing
- [ ] I've updated Swagger to reflect any API changes

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/resources/DataRequestVoteResource.java [hunk 1]
```diff
@@ -31,17 +31,7 @@
...
-import org.broadinstitute.consent.http.service.AbstractElectionAPI;
-import org.broadinstitute.consent.http.service.AbstractVoteAPI;
-import org.broadinstitute.consent.http.service.DataAccessRequestAPI;
-import org.broadinstitute.consent.http.service.DataSetAPI;
-import org.broadinstitute.consent.http.service.DatasetAssociationService;
-import org.broadinstitute.consent.http.service.ElectionAPI;
-import org.broadinstitute.consent.http.service.EmailNotifierService;
-import org.broadinstitute.consent.http.service.VoteAPI;
-import org.broadinstitute.consent.http.service.VoteService;
+import org.broadinstitute.consent.http.service.*;
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/DataRequestVoteResource.java]: nit: Avoid star imports

File: src/main/java/org/broadinstitute/consent/http/service/DatasetService.java [hunk 2]
```diff
@@ -1,22 +1,14 @@
 package org.broadinstitute.consent.http.service;
 
-import java.util.ArrayList;
-import java.util.Collections;
+import java.util.*;
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DatasetService.java]: nit: Avoid star imports here too

File: src/main/java/org/broadinstitute/consent/http/service/DatasetService.java [hunk 3]
```diff
@@ -35,18 +27,59 @@
     private final ConsentDAO consentDAO;
     private final DataSetDAO dataSetDAO;
     private final UserRoleDAO userRoleDAO;
+    private final DataSetAuditDAO dataSetAuditDAO;
+    private final DatasetAssociationDAO datasetAssociationDAO;
     private final UseRestrictionConverter converter;
     public static String datasetName = "Dataset Name";
+    private final String DELETE = "DELETE";;
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DatasetService.java]: nit: rm extra semi-colon

File: src/main/java/org/broadinstitute/consent/http/service/DatasetService.java [hunk 4]
```diff
@@ -264,6 +297,35 @@ public void deleteDataset(Integer datasetId) {
         dataSetDAO.deleteDataSets(idList);
     }
 
+    public void deleteDataset(Integer dataSetId, Integer dacUserId) throws IllegalStateException {
+        try {
+            dataSetDAO.begin();
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DatasetService.java:317]: Not for this PR, but I found that this has problems when actually used. We should be doing this inside a handle instead (see https://jdbi.org/#_handle). Need to file a story once this PR is merged that covers updating this method.

File: src/test/java/org/broadinstitute/consent/http/service/DatasetServiceTest.java [hunk 5]
```diff
@@ -62,7 +66,7 @@ public void setUp() {
     }
 
     private void initService() {
-        datasetService = new DatasetService(consentDAO, datasetDAO, userRoleDAO, useRestrictionConverter);
+        datasetService = new DatasetService(consentDAO, datasetDAO, userRoleDAO, dataSetAuditDAO, datasetAssociationDAO, useRestrictionConverter);
     }
 
     @Test
```
Comment (rushtong) [inline src/test/java/org/broadinstitute/consent/http/service/DatasetServiceTest.java:78]: I'd love to see new tests for all of the methods moved to the service class. We need to incrementally boost our test coverage especially across the service classes.

**Review by rushtong — CHANGES_REQUESTED** [review]
This is fantastic, thank you! Very minor nits inline. Also, we should boost the test coverage as much as possible too.

File: src/main/java/org/broadinstitute/consent/http/service/DatasetService.java (hunk 2 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/DatasetService.java]: It looks like this is still an issue in this file.

**Review by rushtong — APPROVED** [review]
Thank you for the changes! One minor *-import issue and then please feel free to merge 👍🏽

File: src/main/java/org/broadinstitute/consent/http/service/DatasetService.java (hunk 2 above)
Comment (rfricke-asymmetrik) [reply src/main/java/org/broadinstitute/consent/http/service/DatasetService.java]: Missed this one somehow, should be good now.

**Review by rushtong — APPROVED** [review]
Looks great!

**Pull request merged**